src/Network/Email/Envelope.php

```diff
@@ -0,0 +1,65 @@
+<?php
+namespace Cake\Network\Email;
+
+use ArrayAccess;
+
+class Envelope implements ArrayAccess
+{
+    protected $_email;
+
+    public function __construct(Email $email = null)
+    {
+        if ($email === null) {
+            $email = new Email();
+        }
+
+        $this->_email = $email;
+    }
+
+    public function __call($method, $args)
+    {
+        $this[$method] = $args;
```

Comment:
It would be nice if this class provided some conventions around emails too. ActionMailer in ruby does a good job of this.